### PR TITLE
Adding new preview API for promotions

### PR DESCRIPTION
**Pull request by rkapso**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [X] new API version 
      - [ ] update existing version for new feature 
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

To initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by asarkar84** [comment]
Please get an internal approval from ACE API reviewers before going ahead with ARM review.

**Comment by rkapso** [comment]
> Please get an internal approval from ACE API reviewers before going ahead with ARM review.

Will do, this was published to ACE internal review.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json [hunk 1]
```diff
@@ -0,0 +1,9330 @@
...
+          },
+          {
+            "$ref": "#/parameters/billingRoleAssignmentNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/BillingRoleAssignment"
```
Comment (ravbhatnagar) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json:5124]: Delete mustnot return  a response body. It should be empty.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json [hunk 2]
```diff
@@ -0,0 +1,9330 @@
...
+            }
+          },
+          "204": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/BillingRoleAssignment"
```
Comment (ravbhatnagar) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json:5130]: Delete mustnot return a response body. It should be empty. A bunch of 204  response codes are added in this PR all of which have a response body. Existing 200 OK responses also have a body. This is not compliant with Azure Resource Provider contract.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/examples/PromotionActivation.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  "parameters": {
+    "api-version": "2019-11-01-preview",
+    "promotionId": "{promotionId}"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "/providers/Microsoft.Billing/promotions/ff0f1add-122a-43c1-a2ec-c2277781667d",
+        "type": "Microsoft.Billing/promotions",
+        "name": "ff",
```
Comment (ravbhatnagar) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/examples/PromotionActivation.json:11]: name value should match the value in id.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json [hunk 4]
```diff
@@ -0,0 +1,9330 @@
...
+      "put": {
+        "summary": "Activate promotion",
+        "description": "Activate promotion and create resource promotion resource.",
+        "operationId": "Activate_Promotion",
+        "x-ms-examples": {
+          "PromotionActivation": {
+            "$ref": "./examples/PromotionActivation.json"
+          }
+        },
+        "parameters": [
```
Comment (ravbhatnagar) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json:5641]: is there a request body that is needed to create or activate a promotion? Where are things like applied scoped takenn as input?

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json [hunk 5]
```diff
@@ -0,0 +1,9330 @@
...
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Billing/promotions/{promotionId}": {
```
Comment (ravbhatnagar) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json:5631]: what value do the customers need to put in the {promotionId}? Can it be any string or is it a guid? What would happen if they re-PUT with the same value? it should not fail.

Comment (ravbhatnagar) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json:5631]: how is a promotion deactivated or canceled? Delete is missing. If customer can create somethign via a PUT, they shiold be able to delete it via Delete.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json [hunk 6]
```diff
@@ -0,0 +1,9330 @@
...
+          "default": {
+            "description": "Unexpected error",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Billing/promotions/{promotionSkuId}": {
```
Comment (ravbhatnagar) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json:5740]: this way to model eligibility of a subscription is weird and can be confusing.  It looks like a GET API but if the subscription segment is added to the path it transitions to a check eligibility API.
Better way to model will be to take subscription Id in the body. And may be make it a POST /providers/Microsoft.Billing/promotions/{id}/checkEligibility

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json [hunk 7]
```diff
@@ -0,0 +1,9330 @@
...
+          "readOnly": true,
+          "$ref": "#/definitions/Amount"
+        },
+        "subTotal": {
+          "description": "The pre-tax amount due. This field is applicable to billing accounts with agreement type Microsoft Customer Agreement.",
+          "readOnly": true,
+          "$ref": "#/definitions/Amount"
+        },
+        "taxAmount": {
+          "description": "The amount of tax charged for the billing period. This field is applicable to billing accoun
```
Comment (ravbhatnagar) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json:9068]: read only?

Comment (ravbhatnagar) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json:9078]: read only

Comment (ravbhatnagar) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json:9081]: typo? reserveration -> promotion?

Comment (ravbhatnagar) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json:9083]: readonly

**Review by ravbhatnagar — COMMENTED** [review]
@rkapso - added a few comments. please take a look

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json (hunk 5 above)
Comment (rkapso) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json:5631]: There is an expiry on the promotion, so there is no explicit delete.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json (hunk 6 above)
Comment (rkapso) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json:5740]: The reason is we want to check the eligibility on 1000s of such subscriptions that a user may have and leverage the batch semantics that are available only on get.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json (hunk 5 above)
Comment (rkapso) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json:5631]: It is a Guid if the same value is put as input you get the same response it does not fail.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/examples/PromotionActivation.json (hunk 3 above)
Comment (rkapso) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/examples/PromotionActivation.json:11]: ack.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json (hunk 7 above)
Comment (rkapso) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json:9081]: ack

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json (hunk 4 above)
Comment (rkapso) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json:5641]: yes let me add that to the e.g.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json (hunk 1 above)
Comment (rkapso) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json]: this was an accidental change reverted.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json (hunk 6 above)
Comment (rkapso) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json]: Discussed offline added the /checkEligibility

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json (hunk 2 above)
Comment (rkapso) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json]: this was a accidental change reverted.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json (hunk 7 above)
Comment (rkapso) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json]: accidental change reverted.

Comment (rkapso) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json]: accidental change reverted.

Comment (rkapso) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json]: reverted.

Comment (rkapso) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json]: reverted.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json (hunk 5 above)
Comment (rkapso) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json]: discussed offline these so auto expire and show expired in the list/get exp and we expect less than 5 of such per billing account feedback taken added the below work item for DELETE ETA 11/20

https://msazure.visualstudio.com/One/_workitems/edit/8785017

**Comment by ravbhatnagar** [comment]
Signing off from ARM side

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/examples/GetPromotions.json [hunk 8]
```diff
@@ -0,0 +1,27 @@
+{
+  "parameters": {
+    "api-version": "2019-11-01-preview"
```
Comment (asarkar84) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/examples/GetPromotions.json:3]: Is this right version?

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/examples/PromotionCheckEligibility.json [hunk 9]
```diff
@@ -0,0 +1,21 @@
+{
+  "parameters": {
+    "api-version": "2019-11-01-preview",
```
Comment (asarkar84) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/examples/PromotionCheckEligibility.json:3]: Please change the version.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/examples/PromotionDetails.json [hunk 10]
```diff
@@ -0,0 +1,26 @@
+{
+  "parameters": {
+    "api-version": "2019-10-01-preview",
```
Comment (asarkar84) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/examples/PromotionDetails.json:3]: Please change the version.

**Review by asarkar84 — CHANGES_REQUESTED** [review]

**Review by wilcobmsft — APPROVED** [review]

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/examples/GetPromotions.json (hunk 8 above)
Comment (rkapso) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/examples/GetPromotions.json]: fixed

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/examples/PromotionDetails.json (hunk 10 above)
Comment (rkapso) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/examples/PromotionDetails.json]: fixed

File: specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/examples/PromotionCheckEligibility.json (hunk 9 above)
Comment (rkapso) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/examples/PromotionCheckEligibility.json]: fixed

**Comment by asarkar84** [comment]
Approved

**Comment by rkapso** [comment]
> Please get an internal approval from ACE API reviewers before going ahead with ARM review.

done

**Comment by rkapso** [comment]
@xccc-msft can you please merge this

**Review by xseeseesee — APPROVED** [review]

**Review by ravbhatnagar — APPROVED** [review]

**Review by asarkar84 — APPROVED** [review]

**Comment by akning-ms** [comment]
Hi rkapso, I noticed below SDK breaking change looks caused by rename:   Can I know why has this rename for operation_id?
info	[Changelog]   - Removed operation InvoicesOperations.download_multiple_billing_subscription_invoice
info	[Changelog]   - Removed operation InvoicesOperations.download_multiple_modern_invoice

**Comment by akning-ms** [comment]
@rkapso kindly ping for my comment

**Comment by rkapso** [comment]
> Hi rkapso, I noticed below SDK breaking change looks caused by rename: Can I know why has this rename for operation_id?
> info [Changelog] - Removed operation InvoicesOperations.download_multiple_billing_subscription_invoice
> info [Changelog] - Removed operation InvoicesOperations.download_multiple_modern_invoice

The old operation was not used at has been removed

**Comment by akning-ms** [comment]
> > Hi rkapso, I noticed below SDK breaking change looks caused by rename: Can I know why has this rename for operation_id?
> > info [Changelog] - Removed operation InvoicesOperations.download_multiple_billing_subscription_invoice
> > info [Changelog] - Removed operation InvoicesOperations.download_multiple_modern_invoice
> 
> The old operation was not used at has been removed

Thanks @rkapso , my question is why rename it as it caused sdk breaking change which has impact to customer. @msyyc for comments the impact for customer

**Comment by akning-ms** [comment]
and Is this PR still in use. there is lots of validation failure https://github.com/Azure/azure-rest-api-specs/pull/11286
Seems it is adding api in billing.json for same feature. it is not consistent as this PR. @rkapso

**Pull request merged**